ppl/src/main/java/org/opensearch/sql/ppl/utils/ArgumentFactory.java

```diff
@@ -148,11 +151,34 @@
             .singletonList(new Argument("k", getArgumentValue(ctx.k)));
   }
 
+  /**
+   * Get map of {@link Argument}.
+   *
+   * @param ctx ADCommandContext instance
+   * @return the list of arguments fetched from the AD command
+   */
+  public static Map<String, Literal> getArgumentMap(AdCommandContext ctx) {
+    return new HashMap<String, Literal>() {{
+        put("shingle_size", (ctx.shingle_size != null)
+              ? getArgumentValue(ctx.shingle_size)
+              : new Literal(8, DataType.INTEGER));
```

Comment:
If shingle_size = 8 is default setting, it would be better to encapsulated in AD operator instead of in parser.

Reply:
Shingle size is optional, MLCommons will set it as 8 by default https://github.com/opensearch-project/ml-commons/blob/main/ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/rcf/FixedInTimeRandomCutForest.java#L82

Reply:
Changed to not set default value here in the latest revision